examples/waiting-for-goroutines-to-finish/waiting-for-goroutines-to-finish.go

```diff
@@ -0,0 +1,47 @@
+// To wait for multiple goroutines to finish, we can
+// use a sync.WaitGroup.
+
+package main
+
+import (
+    "fmt"
+    "math/rand"
+    "sync"
+    "time"
+)
+
+// This is the function we'll run in every goroutine.
+// wg is the WaitGroup it uses to notify that it's done.
+// Note that a WaitGroup must be passed to functions by
+// pointer.
+func worker(id int, wg *sync.WaitGroup) {
+    fmt.Printf("Worker %d starting\n", id)
+
+    // Sleep for a random duration between 500-700 ms
```

Comment:
Do you feel like the randomization here is essential?

Reply:
Not essential, but for the presentation I think it's useful to not expect things to be deterministic. If all workers waited for time T, main could have just `time.Sleep(T+eps)` too wait for them, so the example is less convincing.

We could do something alternative like issue some HTTP HEAD requests to a few common URLs, instead.

I don't feel strongly about this, if you think we could replace the sleeps by:

```
// ... do work here
```

I'm OK with that too.

Reply:
I went ahead and changed that to just `time.Sleep(time.Second)` to match what the worker pool sample is doing. Consistency is important